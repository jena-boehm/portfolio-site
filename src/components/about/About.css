html {
    scroll-behavior: smooth;
}

.aboutContainer {
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-direction: column;
    height: max-content;
    width: 100%;
    box-sizing: border-box;
    font-family: 'Montserrat', sans-serif;
    color: white;

    .titleHome {
        font-size: 3.5em;
        letter-spacing: 0.1em;
        opacity: 0.4;
        margin-bottom: 0;
        display: none;
    }

    .titleLeft {
        transform: rotate(270deg);
        font-size: 8em;
        letter-spacing: 0.1em;
        margin: 0;
        padding: 0;
        height: max-content;
        opacity: 0.3;
        position: fixed;
        left: -11vw;
        top: 44vh;
        z-index: -1;
        font-weight: 300;
        color: #B0B6FB;
        text-shadow: 6px 6px 1px black;
    }

    .aboutRight {
        display: flex;
        justify-content: center;
        align-items: center;
        flex-direction: column;
        height: 100%;
        width: 80%;
        padding: 0;

        .aboutRightSection {
            display: flex;
            justify-content: center;
            align-items: center;
            flex-direction: column;

            .rightTop {
                width: 100%;
                display: flex;
                justify-content: center;
                align-items: center;
                flex-direction: column;
                    
                    .crop {
                        width: max-content;
                        height: 50vh;
                        width: 60vw;
                        position: relative;
                        overflow: hidden;
                        background-image: url('../../../public/images/headshot.jpg');
                        background-size: cover;
                        background-repeat: no-repeat;
                        background-position: top;
                        /* border-radius: 50%; */
                        box-shadow: rgba(0, 0, 0, 0.4) 0px 50px 100px -20px, 
                        rgba(0, 0, 0, 0.7) 0px 30px 60px -30px, 
                        rgba(0, 0, 0, 0.7) 0px -2px 6px 0px inset;

                        
                        /* .headshot {
                            display: inline-block;
                            margin-top: -3%;
                            height: 130%;
                            width: auto;
                            opacity: 0.9;
                        } */
                    }

                .description {
                    font-size: 1.3em;
                    margin-top: 5vh;
                }
            }

            .rightMain {
                width: 100%;
                height: 60vh;
                display: flex;
                justify-content: center;
                align-items: center;
                flex-direction: row;
                margin: 2em 0 2em 0;
                border: 1px solid rgba(176, 182, 251, 0.2);
                box-shadow: rgba(176, 182, 251, 0.3) 0px 2px 5px 1px, rgba(176, 182, 251, 0.23) 0px 2px 6px 1px, rgba(0, 0, 0, 0.4) 0px 50px 100px -20px, 
                rgba(0, 0, 0, 0.7) 0px 30px 60px -30px, 
                rgba(0, 0, 0, 0.7) 0px -2px 6px 0px inset;

                .bio {
                    width: 45%;
                    line-height: 2em;
                    text-align: justify;
                    padding: 0 1em 0 1em;
                }

                .line {
                    height: 90%;
                    width: 5%;
                    background-image: url('../../../public/images/vertical-line.png');
                    background-repeat: repeat-y;
                    background-position: center;
                    background-size: 10%;
                    opacity: 0.5;
                }

                .techStack {
                    display: flex;
                    justify-content: center;
                    align-items: stretch;
                    width: 45%;
                    height: max-content;
                    line-height: 1.5em;
                    text-align: center;
                    padding: 0 0 1.1em 0;

                    .techStackTitle {
                        text-align: center;
                        text-decoration: underline;
                    }

                    .techStackLeft {
                        display: flex;
                        justify-content: center;
                        align-content: space-between;
                        flex-direction: column;
                        width: 50%;
                        height: 100%;
                        padding-right: 0.5em;

                        .stackHeader {
                            font-size: 0.7em;
                            text-transform: uppercase;
                            letter-spacing: 0.2em;
                            margin-top: 2.1em;
                            font-weight: 600;
                            color: #B0B6FB;
                        }
                    }

                    .techStackRight {
                        display: flex;
                        justify-content: center;
                        align-content: space-between;
                        flex-direction: column;
                        width: 50%;
                        height: 100%;
                        padding-right: 0.5em;

                        .stackHeader {
                            font-size: 0.7em;
                            text-transform: uppercase;
                            letter-spacing: 0.2em;
                            margin-top: 2.1em;
                            font-weight: 600;
                            color: #B0B6FB;
                        }
                    }
                }
            }
    }
        .bottomBlurb {
            width: 70%;
            text-align: center;
            line-height: 1.5em;
            margin-top: 3em;
            
        }

        .socials {
            display: flex;
            justify-content: center;
            align-items: center;
            margin: 6vh 0 2vh 0;
            opacity: 0.8;
        }

        .scroll {
            cursor: pointer;
            margin-bottom: 3em;
            position: relative;
            top: 45%;
            margin: 2vh 0 4vh 0;
            cursor: pointer;
        }
    }
}

@media (min-width: 1500px) {
    .aboutContainer {

        .titleLeft {
            left: -11vw;
            font-size: 10em;
        }

        .aboutRight {

            .bottomBlurb {
                margin-bottom: 4vh;
            }

            .scroll {
                margin: 4vh 0 6vh 0;
            }
        }
    }
} 