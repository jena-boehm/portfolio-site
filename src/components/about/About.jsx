/* eslint-disable max-len */
import React, { useRef, useState, useEffect } from 'react';
import { useLocation } from 'react-router-dom';
import styles from './About.css';
// import headshot from '../../../public/images/headshot.jpg';
import { Socials } from '../socials/Socials';
import { ScrollTwo } from '../scroll/ScrollTwo';

export const About = () => {
  const location = useLocation();
  const ref = useRef(null);
  const [isMobile, setMobile] = useState(window.innerWidth < 1100);

  const handleScroll = () => {
    ref.current.scrollIntoView();
  };

  const handleMobile = () => {
    setMobile(window.innerWidth < 1100);
  };

  useEffect(() => {
    window.addEventListener('resize', handleMobile);
    return () => window.removeEventListener('resize', handleMobile);
  });

  return (
    <div className={styles.aboutContainer}>
      <h1
        className={styles.titleLeft}
        style={
          location.pathname === '/'
            ? { display: 'none' }
            : isMobile
            ? { display: 'none' }
            : { display: 'flex' }
        }
      >
        ABOUT
      </h1>

      <section
        className={styles.aboutRight}
        style={
          location.pathname === '/about' && isMobile
            ? { margin: '2em' }
            : location.pathname === '/'
            ? { margin: '2em' }
            : { margin: '2em 2em 2em 13em' }
        }
      >
        <section className={styles.aboutRightSection}>
          <section className={styles.rightTop}>
            {location.pathname === '/about' && isMobile && (
              <div className={styles.titleTop}>ABOUT</div>
            )}
            <div className={styles.crop}>
              {/* <img
                className={styles.headshot} 
                src={headshot}
                alt="Jena headshot"
                height="50%"
                width="auto"/> */}
            </div>

            <div className={styles.description}>
              Multifaceted, intuitive, and detail-driven Front End Software
              Engineer.
            </div>

            <section className={styles.socials}>
              <Socials />
            </section>

            <section className={styles.scroll}>
              <div onClick={handleScroll}>
                <ScrollTwo />
              </div>
            </section>
          </section>

          <section className={styles.rightMain} ref={ref}>
            <div className={styles.bio}>
              My work is inspired by open communication, authentic expression,
              and effective organization. Experienced with Full Stack, but front
              end really feels like home.
              <br />
              <br />
              With a knack for creative problem-solving and an eye for front end
              design, I thrive in user-focused environments that fuse technology
              and art together in innovative ways.
              <br />
              <br />
              Tech is the future, and I believe that as a software engineer I am
              able to directly and positively impact the lives of many people.
            </div>

            <div className={styles.line}></div>

            <div className={styles.techStack}>
              {/* <div className={styles.techStackTitle}>
              Tech Stack<br/><br/>
              </div> */}
              <div className={styles.techStackLeft}>
                <div className={styles.stackItem}>
                  <div className={styles.stackHeader}>Languages:</div>
                  <div>HTML, CSS, JavaScript ES6</div>
                </div>

                <div className={styles.stackItem}>
                  <div className={styles.stackHeader}>
                    Libraries & Frameworks:
                  </div>
                  <div>React, Next.js, Gatsby, Node.js, GraphQL, Express</div>
                </div>

                <div className={styles.stackItem}>
                  <div className={styles.stackHeader}>Testing:</div>
                  <div>Jest, QUnit, Supertest, React Testing Library</div>
                </div>

                <div className={styles.stackItem}>
                  <div className={styles.stackHeader}>Databases:</div>
                  <div>PostgreSQL</div>
                </div>
              </div>

              <div className={styles.techStackRight}>
                <div className={styles.stackItem}>
                  <div className={styles.stackHeader}>Tools:</div>
                  <div>
                    VSCode, Git, GitHub, Heroku, Netlify, Postman, PGAdmin,
                    Ableton Live
                  </div>
                </div>

                <div className={styles.stackItem}>
                  <div className={styles.stackHeader}>
                    Project Management Tools:
                  </div>
                  <div>Miro, Figma, Trello, Airtable, Asana, Jira</div>
                </div>

                <div className={styles.stackItem}>
                  <div className={styles.stackHeader}>Dev Practices:</div>
                  <div>
                    Pair Programming, RESTful APIs, Agile, Test Driven
                    Development, Design Driven Development
                  </div>
                </div>
              </div>
            </div>
          </section>
        </section>

        <section className={styles.bottomBlurb}>
          Other passions and skills:
          <br />
          Electronic music production, audio engineering, DJing & live
          performing, event production, transcribing, volunteer coordination,
          dance, and classical piano.
        </section>
      </section>
    </div>
  );
};
