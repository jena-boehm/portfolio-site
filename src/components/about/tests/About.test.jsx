import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { About } from '../About';

describe('About component', () => {
  afterEach(() => cleanup());
  test.skip('renders About', () => {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });
});
