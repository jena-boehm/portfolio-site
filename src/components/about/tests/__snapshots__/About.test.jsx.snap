// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders About 1`] = `
<DocumentFragment>
  <div>
    About
  </div>
</DocumentFragment>
`;
